ephios/core/services/notifications/backends.py

```diff
@@ -25,25 +30,55 @@ def enabled_notification_backends():
         yield from (b() for b in backends)
 
 
+def send_a_notification(backend):
+    with transaction.atomic():
+        # grab an untried notification and mark it
+        if (
+            notification := Notification.objects.exclude(backends_tried__has_key=backend.slug)
+            .exclude(backends_tried__has_key=backend.slug)
+            .first()
+        ) is None:
+            return
+        notification.backends_tried[backend.slug] = False
+        notification.save()
+
+    # send it
+    if backend.can_send(notification) and backend.user_prefers_sending(notification):
+        try:
+            backend.send(notification)
+        except Exception as e:  # pylint: disable=broad-except
+            if settings.DEBUG:
+                raise e
+            message = f"Notification {notification.pk}: {notification}\nException: {e}\n{traceback.format_exc()}"
```

Comment:
it would be great to have the SITE_URL in here as you cannot distinguish the source of the error by the sender anymore. currently this is only possible via the path in the stacktrace